In [51]:
import math
import matplotlib.pyplot as plt

Calculating $e$, eulers constant

$e = \sum_{n=1}^{\infty} \; (\frac{1}{n!})$

In [52]:
def factorial(n: int) -> int:
    sum: int = 1
    for i in range(1, n + 1):
        sum *= i
    return sum

def e(precision: int) -> float:
    sum: float = 1
    for n in range(1, precision + 1):
        sum += 1 / factorial(n)
    return sum

e(20)

2.7182818284590455

$v = p * e^{rt}$ where $v$ is the final amount of money, $e$ is eulers constant, $r$ is the rate of interest and $t$ is the time.
 
We can manipulate this formula algebraically to solve for $t$: 

$\frac{v}{p} = e^{rt}$
 
$\ln (\frac{v}{p}) = \ln(e^{rt})$
 
$\ln (\frac{v}{p}) = rt * \ln(e)$
Since $\ln(e)$ equals $e^x = e$ and when $x = 1$ they are equal, we can simplify $\ln(e)$ to $1$
 
$\ln (\frac{v}{p}) = rt * 1 = rt$
 
$\frac{ln (\frac{v}{p})}{r} = t$

In [53]:
# Calculates when we're going to be rich.
def when_rich(r: float, v: float, p: float) -> float:
    numerator: float = math.log((v / p), math.e)
    return numerator / r


In [54]:
when_rich(0.05, 100000, 10)

184.20680743952366

11. $16^2 \cdot 16^{-1.75} = 16^{2 - 1.75} = 16^{0.25} = 16^{\frac{1}{4}} = \sqrt[4]{16} = 2$
---
12. $9^{1/3} \dot 9^{1/6} = 9^{1/3 + 1/6} = 9^{2/6 + 1/6} = 9^{3/6} = 9^{1/2} = \sqrt[2]{9} = 3$
---
13. $\frac{4^{4.2}}{4^{3.7}} = 4^{4.2 - 3.7} = 4^{0.5} = 4^{\frac{1}{2}} = \sqrt[2]{4} = 2$
---
14. $\frac{3^{5/3}}{3^{2/3}} = 3^{5/3 - 2/3} = 3^{3/3} = 3^1 = 3$
---
15. $(25^{1/8})^4 = 25^{4/8} = 25^{1/2} = \sqrt[2]{25} = 5$
---
16. $\big (13^{\sqrt{2}} \big)^{\sqrt{2}/2} = 13^{\sqrt{2} \cdot \sqrt{2}/2} = 13^{2/2} = 13^{1} = 13$
---
17. $2^{\sqrt{3}} \cdot 7^{\sqrt{3}} = \big( 2 \cdot 7 \big)^{\sqrt{3}} = 14^{\sqrt{3}}$
---
18. $\big( \sqrt{3} \big)^{1/2} \cdot \big( \sqrt{12} \big)^{1/2} = \big( \sqrt{3} \cdot \sqrt{12} \big)^{1/2} = 6^{1/2} = \sqrt{6}$

    $\sqrt{3} \cdot \sqrt{12} = \sqrt{3 \cdot 12} = \sqrt{36} = 6$
---
19. $ \big( \frac{2}{\sqrt{2}} \big)^4 = \big( \frac{2^4}{(\sqrt{2})^4} \big) = \frac{16}{4} = 4$

    $2^4 = 16, \sqrt{2}^4 = 2 * 2 = 4$
---
20. $ \big( \frac{\sqrt{6}}{3} \big)^2 = \frac{6}{9} = \frac{2}{3} $
---

21. $f(x) = \frac{1}{2 + e^x}$

    $Domain: \{x \; \epsilon \; \real \}, \; Range: \{f(x) \; \epsilon \; \real \; | \; f(x) \; \gt \; 0 \}$
---
22. $g(t) = cos(e^{-t})$
    
    $Domain: \{x \; \epsilon \; \real \}, \; Range: \{g(x) \; \epsilon \; \real \; | \; -1 \; \leq \; g(x) \; \leq \; 1 \}$
---
23. $g(t) = \sqrt{1 + 3^{-t}}$

    $Domain: \{t \; \epsilon \; \real \}, \; Range: \{g(t) \; \epsilon \; \real \; | \; g(t) \; \gt \; 1 \}$
---
24. $f(x) = \frac{3}{1 - e^{2x}}$

    $Domain: \{x \; \epsilon \; \real \}, \; Range: \{g(x) \; \epsilon \; \real \; | \; g(x) \; \lt \; 3 \}$
---

25.
    $2^x = 5$

    $\log_2(2^x) = \log_2(5)$

    $x \cdot \log_2(2) = \log_2(5)$

    $x = \log_2(5) = \log_2(2 \cdot 2.5) = \log_2(2) + \log_2(2.5) = 1 + \log_2(2.5)$
---

29. $p_b = 500,000, \; p_d = 1,000,000$ where $p_b$ is the population in the beginning, and $p_d$ is the population we will reach. The population grows by $3.75\%$ each year, which we will indicate by $r = 0.0375$.

    $p_d = p_b \cdot e^{rt}$

    Now we want to solve this equation for t, which represents the number of years until we reach the population.

    $\frac{p_d}{p_b} = e^{rt}$

    $\ln{(\frac{p_d}{p_b})} = \ln{(e^{rt})} = rt \cdot \ln{(e)} = rt$

    $\frac{\ln{(\frac{p_d}{p_b})}}{r} = t$

In [55]:
def assignment_29() -> None:
    p_b, p_d, r = 500_000, 1_000_000, 0.0375
    t = math.log(p_d / p_b, math.e) / r
    print(t)

assignment_29()

18.483924814931875


31. $f_i$ is the initial amount of phosphorus-32, and $f_t$ is the amount of phosphorus-32 at time $t$. Initially there will be $6.6g$ present, which we will be put in $f_i = 6.6$. We will indicate the rate of decay by $r$ however this is unknown! We need to solve for it first. For this we will solve the exponential growth equation for $r$.

$f_t = f_i \cdot e^{rt}$

$\frac{f_t}{f_i} = e^{rt}$

$\ln{(\frac{f_t}{f_i})} = \ln{(e^{rt})}$

$\ln{(\frac{f_t}{f_i})} = rt \cdot \ln{e} = rt$

$\frac{\ln{(\frac{f_t}{f_i})}}{t} = r$

Now we have this, let's calculate $r$ with the given values: $f_t = 3.3, \; f_i = 6.6, \; t = 14$

r = $\frac{\ln{(\frac{3.3}{6.6})}}{14}$

Let's evaluate this bellow:

In [56]:
math.log(3.3 / 6.6, math.e) / 14

-0.049510512897138946

Now we know that $r \approx -0.04951$, so now we can express the amount of phosphorus-32 as: $f(t) = 6.6 \cdot e^{-0.04951t}$

With all of this information, and the previously derived equation (slightly modified) $\frac{\ln{(\frac{f_t}{f_i})}}{r} = t$ we can figure out how long it will take to reach 1g.

$\frac{\ln{(\frac{1}{6.6})}}{-0.04951} = t$

In [57]:
math.log(1/6.6, math.e) / -0.04951

38.114919188696824

Now we have $t$, which tells us that it will take $\approx 38.11492$ days to reach 1g of phosphorus.

---

33. We want to know how much time it would take for an investment to double when the interest rate is $6.25\%$. We define $v$ as the value we want to double, $w = 2v$ and $r = 0.0625$

$\frac{\ln{(\frac{w}{v})}}{6.25} = \frac{\ln{(\frac{2v}{v})}}{6.25} = \frac{\ln{2}}{6.25} = t$

In [59]:
math.log(2, math.e) / 0.0625

11.090354888959125

We now determined that it would take $\approx 11$ years to double, no matter what the original value is.
